packages/block-directory/src/components/downloadable-blocks-list/index.js

```diff
@@ -50,65 +50,57 @@ function DownloadableBlocksList( { items, onHover = noop, children, downloadAndI
 }
 
 export default compose(
+	withSelect( ( select ) => {
+		const {
+			getErrorNotices,
+			isInstalling,
+		} = select( 'core/block-directory' );
+
+		const errorNotices = getErrorNotices();
+		const isLoading = isInstalling();
+
+		return {
+			errorNotices,
+			isLoading,
+		};
+	} ),
 	withDispatch( ( dispatch, props ) => {
-		const { installBlock, downloadBlock } = dispatch( 'core/block-directory' );
-		const { createErrorNotice, removeNotice } = dispatch( 'core/notices' );
-		const { removeBlocks } = dispatch( 'core/block-editor' );
+		const { downloadBlock, installBlock, setErrorNotice, clearErrorNotice } = dispatch( 'core/block-directory' );
 		const { onSelect } = props;
 
-		return {
-			downloadAndInstallBlock: ( item ) => {
-				const onDownloadError = () => {
-					createErrorNotice(
-						__( 'Block previews can’t load.' ),
-						{
-							id: DOWNLOAD_ERROR_NOTICE_ID,
-							actions: [
-								{
-									label: __( 'Retry' ),
-									onClick: () => {
-										removeNotice( DOWNLOAD_ERROR_NOTICE_ID );
-										downloadBlock( item, onSuccess, onDownloadError );
-									},
-								},
-							],
-						} );
-				};
+		const download = ( item ) => {
+			clearErrorNotice( item.id );
 
-				const onSuccess = () => {
-					const createdBlock = onSelect( item );
+			const onDownloadError = () => {
+				setErrorNotice( item.id, DOWNLOAD_ERROR_NOTICE_ID );
+			};
 
-					const onInstallBlockError = () => {
-						createErrorNotice(
-							__( 'Block previews can\'t install.' ),
-							{
-								id: INSTALL_ERROR_NOTICE_ID,
-								actions: [
-									{
-										label: __( 'Retry' ),
-										onClick: () => {
-											removeNotice( INSTALL_ERROR_NOTICE_ID );
-											installBlock( item, noop, onInstallBlockError );
-										},
-									},
-									{
-										label: __( 'Remove' ),
-										onClick: () => {
-											removeNotice( INSTALL_ERROR_NOTICE_ID );
-											removeBlocks( createdBlock.clientId );
-											unregisterBlockType( item.name );
-										},
-									},
-								],
-							}
-						);
-					};
+			const onDownloadSuccess = () => {
+				onSelect( item );
+			};
 
-					installBlock( item, noop, onInstallBlockError );
+			downloadBlock( item, onDownloadSuccess, onDownloadError );
+		};
+
+		const install = ( item, onSuccess ) => {
+			clearErrorNotice( item.id );
+
+			const onInstallBlockError = () => {
+				setErrorNotice( item.id, INSTALL_ERROR_NOTICE_ID );
+			};
+
+			installBlock( item, onSuccess, onInstallBlockError );
+		};
+
+		return {
+			installAndDownload: ( item ) => {
```

Comment:
We can use object function notation to remove some syntax here.

```suggestion
			installAndDownload( item ) {
```